tests/test_db_store.py

```diff
@@ -34,6 +34,20 @@ def test_latest_soil_moisture_empty_database(self):
         moisture = store.latest_soil_moisture()
         self.assertIsNone(moisture)
 
+    def test_retrieve_soil_moisture(self):
+        mock_cursor = mock.Mock()
+        store = db_store.SoilMoistureStore(mock_cursor)
+        timestamp1 = str(datetime.datetime(2016, 7, 23, 10, 51, 9, 928000))
+        timestamp2 = str(datetime.datetime(2016, 7, 23, 10, 52, 9, 928000))
+        mock_cursor.fetchall.return_value = [(timestamp1, 300),
+                                             (timestamp2, 400)]
+        soil_moisture_data = store.retrieve_soil_moisture()
+
+        self.assertEqual(soil_moisture_data[0].timestamp, timestamp1)
+        self.assertEqual(soil_moisture_data[0].soil_moisture, 300)
+        self.assertEqual(soil_moisture_data[1].timestamp, timestamp2)
+        self.assertEqual(soil_moisture_data[1].soil_moisture, 400)
```

Comment:
One subtlety here is that the test assumes an ordering or the result list, but the function contract doesn't guarantee one. The actual implementation is that the order is just the same as the order from sqlite, but that's an implementation detail and the test shouldn't depend on that.

Normally if we wanted to compare lists without caring about order, we could just use `assertItemsEqual`, but it's a pain because I don't think our `namedtuple`s implement `equals` the way we want.

I think that leaves either enforcing ordering as part of the function contract (e.g. "returns soil moisture and timestamp in chronological order") or leave the function as is and just sort the results in the test so we're guaranteed an ordering.